### PR TITLE
feat: add EXECUTE IMMEDIATE statement

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Related https://github.com/datafuselabs/databend/issues/14904
Closes https://github.com/datafuselabs/databend/issues/15200

Adding `EXECUTE IMMEDIATE` command which is used to execute a SQL script.

Example:

```sql
EXECUTE IMMEDIATE $$
BEGIN
    LET x := 1;
    LET y := x + 1;
    LET z RESULTSET := SELECT :y + 1;
    RETURN TABLE(z);
END;
$$;
----
┌───────┐
│ 2 + 1 │
│ UInt8 │
├───────┤
│     3 │
└───────┘
```

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15144)

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_execute_immediate.rs [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        };
+
+        res.map_err(|err| err.display_with_sql(&self.plan.script))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct QueryResult {
+    schema: DataSchemaRef,
+    block: DataBlock,
```
Comment (sundy-li) [inline src/query/service/src/interpreters/interpreter_execute_immediate.rs:138]: why not use `blocks:  Vec<DataBlock>`

File: src/query/service/src/interpreters/interpreter_execute_immediate.rs [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+
+        let mut planner = Planner::new(ctx.clone());
+        let (plan, _) = planner.plan_sql(query).await?;
+        let interpreter = InterpreterFactory::get(ctx.clone(), &plan).await?;
+        let stream = interpreter.execute(ctx.clone()).await?;
+        let blocks = stream.try_collect::<Vec<_>>().await?;
+        let schema = plan.schema();
+
+        let block = match blocks.len() {
+            0 => DataBlock::empty(),
```
Comment (sundy-li) [inline src/query/service/src/interpreters/interpreter_execute_immediate.rs:164]: empty_with_schema

File: src/query/script/tests/it/testdata/script-error.txt [hunk 3]
```diff
@@ -9,39 +9,51 @@ error:
...
 ---------- Output ----------
 error: 
-  --> SQL:2:5
+  --> SQL:2:10
   |
-1 | LET x := 1;
-2 | LET x RESULTSET := SELECT 1;
-  |     ^ `x` is already defined as a different kind of variable
+1 | LET x RESULTSET := SELECT 1;
+2 | LET y := x;
```
Comment (sundy-li) [inline src/query/script/tests/it/testdata/script-error.txt:19]: x is `ResultSet`, so y could be `ResultSet` too ?

ResultSet could be cloned.

File: src/query/service/src/interpreters/interpreter_execute_immediate.rs [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+                    let rendered_table = box_render(
+                        &set.schema,
+                        &[set.block.clone()],
+                        usize::MAX,
+                        usize::MAX,
+                        usize::MAX,
+                        true,
+                    )?;
+                    let lines = rendered_table.lines().map(|x| x.to_string()).collect();
+                    PipelineBuildResult::from_blocks(vec![DataBlock::new_from_columns(vec![
```
Comment (sundy-li) [inline src/query/service/src/interpreters/interpreter_execute_immediate.rs:121]: The output result is a single-string column now.

Can we infer the schema before execution?

File: src/query/script/tests/it/testdata/script-error.txt (hunk 3 above)
Comment (andylokandy) [reply src/query/script/tests/it/testdata/script-error.txt:19]: y can be resultset, as long as y is declared as `LET y RESULTSET := x`

File: src/query/service/src/interpreters/interpreter_execute_immediate.rs (hunk 4 above)
Comment (andylokandy) [reply src/query/service/src/interpreters/interpreter_execute_immediate.rs:121]: No, there is no way to infer the schema. Unlike EXECUTE IMMEDIATE, CREATE PROCEDURE must explicitly declare its returning schema.

File: src/query/service/src/interpreters/interpreter_execute_immediate.rs (hunk 1 above)
Comment (andylokandy) [reply src/query/service/src/interpreters/interpreter_execute_immediate.rs:138]: if so, the iter will be more complicated. In most cases, we will only get one block returned, right?

File: src/query/script/tests/it/testdata/script-error.txt (hunk 3 above)
Comment (sundy-li) [reply src/query/script/tests/it/testdata/script-error.txt:19]: Ok, so the default type is Scalar.

**Review by sundy-li — APPROVED** [review]

File: src/query/service/src/interpreters/interpreter_execute_immediate.rs (hunk 1 above)
Comment (sundy-li) [reply src/query/service/src/interpreters/interpreter_execute_immediate.rs:138]: It's right.  We can improve it later.

**Pull request merged**